### PR TITLE
Ensure compatibility for `xarray.NamedArray`

### PR DESCRIPTION
**Pull request by hendrikmakait**
Addresses https://github.com/dask/dask/pull/11166#pullrequestreview-2111244098

`xarray.NamedArray` has been exposed in https://github.com/pydata/xarray/pull/8380/files#diff-1f152a1e45ea432381b82dd02400dbf5b8c4b47d72268cd05d8d4cf6311301b7R44

### CODE REVIEW

File: dask/sizeof.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import sys
 from array import array
 
+from packaging.version import Version
```
Comment (jrbourbeau) [inline dask/sizeof.py:9]: Non-blocking comment: It's strange to have both this line and the `parse_version` import below. We were using the `from packaging.version import parse as parse_version` convention below throughout the codebase, but FWIW I actually like `from packaging.version import Version` better.

**Review by jrbourbeau — COMMENTED** [review]
Thanks @hendrikmakait. This LGTM if CI is happy. GitHub actions are down at the moment https://www.githubstatus.com/incidents/lfrlwdg67fn8 so we'll want to bump CI when things are back online

**Review by jrbourbeau — APPROVED** [review]

File: dask/sizeof.py (hunk 1 above)
Comment (hendrikmakait) [reply dask/sizeof.py:9]: Good catch, I didn't even notice. I've mostly dealt with this in `dataframe/_compat.py` where `Version` appears to be the convention.

Comment (jrbourbeau) [reply dask/sizeof.py:9]: Switching to `Version` was pretty easy with a simple grep/copy/paste over in https://github.com/dask/dask/pull/11171

File: dask/sizeof.py [hunk 2]
```diff
@@ -264,12 +265,17 @@ def register_xarray():
 
     import xarray as xr
 
+    XARRAY_VERSION = Version(xr.__version__)
+    XARRAY_GE_2024_02 = XARRAY_VERSION >= Version("2024.02.0")
+
     @sizeof.register(xr.DataArray)
-    @sizeof.register(xr.NamedArray)
```
Comment (dcherian) [inline dask/sizeof.py:268]: Sorry for the delay here. Using `Variable` here might be enough? (Variable is a subclass of the new NamedArray, and if you need NamedArray, then you need Variable on older versions).

**Pull request merged**